python/lsst/daf/butler/_quantum_backed.py

```diff
@@ -371,12 +378,12 @@ class QuantumProvenanceData(pydantic.BaseModel):
     # `~CollectionType.RUN` level, such as the compute node ID). but adding it
     # now is out of scope for this prototype.
 
-    predicted_inputs: Set[uuid.UUID]
+    predicted_inputs: Set[DatasetId]
```

Comment:
The `QuantumBackedButler` concept doesn't work if we don't have UUIDs (we need to generate dataset IDs in advance, without inserting them into the repo up front, and also without having clashes), so this really can be assumed to be UUID.  I'm opening to changing it later if this is primarily a way to satisfy MyPy for now, but we shouldn't be under any illusions about that.

Reply:
This is indeed to make mypy happy. Clearly this cannot work with integers, but I think we'll be switching to UUIDs eventually and `DatasetId` will be an alias for UUID. I can also add runtime check for the type.

Reply:
Thinking again, I don't want to check the type of every `DatasetRef.id`, there are too many of them. As currently implemented, this butler can only work with the registries that use UUID for `dataset_id` and it will most likely crash if integers are used. I think this should be sufficient protection.